mypy/test/teststubgen.py

```diff
@@ -906,6 +931,50 @@ class TestClass(argparse.Action):
         assert_equal(output, ["class C(argparse.Action): ..."])
         assert_equal(gen.get_imports().splitlines(), ["import argparse"])
 
+    @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
+    def test_inline_generic_class(self) -> None:
+        T = TypeVar("T")
+
+        class TestClass:
```

Comment:
I would also suggest to add a test case with `class Some[T]` test hidden behind `exec()` (for correct syntax) and `skipIf`.

Reply:
Do you mean something like the following?

```python
 @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
    def test_generic_class(self) -> None:

        class TestClass[T]:
            pass
        output: list[str] = []
        mod = ModuleType("module", "")
        gen = InspectionStubGenerator(mod.__name__, known_modules=[mod.__name__], module=mod)
        gen.generate_class_stub("C", TestClass, output)
        assert_equal(output, ["class C[T]: ..."])
        ```

Reply:
Follow up question should this test still live in stubgenc or should it move elsewhere to a different stubgen section?

Reply:
Something like:

```python
    @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
    def test_generic_class(self) -> None:
        exec("class Test[A]: ...")
        ...
```

Reply:
This should now be address by the new commit